### PR TITLE
driver: Add GUD for Keysight/Agilent/HP wfm gens

### PR DESCRIPTION
**Pull request by WilliamHPNielsen**
Since these instruments basically have not changed through three vendors (and at least one decade), they might as well have a common driver. This will also allow for seamless exchange of one unit with another.

Changes proposed in this pull request:
- Add GUD

I have not yet put in deprecation warnings nor created shim drivers pointing to this one. I'll do so if everyone likes the idea.

@jenshnielsen

### CODE REVIEW

File: qcodes/instrument_drivers/Keysight/KeysightAgilent_33XXX.py [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+                           vals=vals.Enum('CW', 'LIST', 'SWEEP', 'FIXED')
+                           )
+
+        self.add_parameter('frequency',
+                           label='Channel {} frequency'.format(channum),
+                           set_cmd='SOURce{}:FREQuency {{}}'.format(channum),
+                           get_cmd='SOURce{}:FREQuency?'.format(channum),
+                           get_parser=float,
+                           unit='Hz',
+                           # TODO: max. freq. actually really tricky
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/Keysight/KeysightAgilent_33XXX.py]: Does this need to depend on the model? I don't know why it's really tricky (does it perhaps depend on output type?)

Comment (WilliamHPNielsen) [reply qcodes/instrument_drivers/Keysight/KeysightAgilent_33XXX.py]: It depends on model (I just pushed that change), but also other things which I don't remember. Amplitude, prolly.

Comment (jenshnielsen) [reply qcodes/instrument_drivers/Keysight/KeysightAgilent_33XXX.py]: never mind

**Comment by WilliamHPNielsen** [comment]
@Dominik-Vogel, maybe you'd like to review this?

**Comment by Dominik-Vogel** [comment]
It looks good to me, great that to made this unification. At some point in the future we are going to be really happy about it!

**Pull request merged**